### PR TITLE
rgw: set correct requestId and hostId on s3select error

### PR DESCRIPTION
**Pull request by clwluvw**
Previously, these fields remained constant despite the possibility of populating them with appropriate values.

Fixes: https://tracker.ceph.com/issues/65468

### CODE REVIEW

**Comment by cbodley** [comment]
@galsalomon66 could you please help with review

**Comment by galsalomon66** [comment]
Hi @clwluvw 
please review https://github.com/ceph/ceph/pull/56834 (it fixes issues related to broken-connection. open BZ's for 7.1)

your changes could be applied on top of that PR.
is that OK?

File: src/rgw/rgw_s3select.cc [hunk 1]
```diff
@@ -246,8 +245,9 @@ void aws_response_handler::send_error_response_rgw_formatter(const char* error_c
   s->formatter->open_object_section("Error");
   s->formatter->dump_string("Code", error_code);
   s->formatter->dump_string("Message", error_message);
-  s->formatter->dump_string("Resource", "#Resource#");
-  s->formatter->dump_string("RequestId", resource_id);
+  if (!s->trans_id.empty())
```
Comment (galsalomon66) [inline src/rgw/rgw_s3select.cc:248]: :+1:

File: src/rgw/rgw_s3select.cc [hunk 2]
```diff
@@ -429,7 +429,7 @@ int RGWSelectObj_ObjStore_S3::run_s3select_on_csv(const char* query, const char*
 
   if (s3select_syntax.get_error_description().empty() == false) {
     //error-flow (syntax-error)
-    m_aws_response_handler.send_error_response_rgw_formatter(s3select_syntax_error,s3select_syntax.get_error_description().c_str(),s3select_resource_id);
+    m_aws_response_handler.send_error_response_rgw_formatter(s3select_syntax_error,s3select_syntax.get_error_description().c_str());
```
Comment (galsalomon66) [inline src/rgw/rgw_s3select.cc:432]: please review https://github.com/ceph/ceph/pull/56834
this part was changed. 
your changes could easily applied to that PR.

**Comment by clwluvw** [comment]
> Hi @clwluvw please review #56834 (it fixes issues related to broken-connection. open BZ's for 7.1)
> 
> your changes could be applied on top of that PR. is that OK?

Hi @galsalomon66, Agreed. no problem. thanks.

**Comment by cbodley** [comment]
> > Hi @clwluvw please review #56834 (it fixes issues related to broken-connection. open BZ's for 7.1)
> > your changes could be applied on top of that PR. is that OK?
> 
> Hi @galsalomon66, Agreed. no problem. thanks.

https://github.com/ceph/ceph/pull/56834 merged so this should be ready for rebase

File: src/rgw/rgw_s3select.cc [hunk 3]
```diff
@@ -214,10 +214,10 @@ void aws_response_handler::init_end_response()
   rgw_flush_formatter_and_reset(s, s->formatter);
 }
 
-void aws_response_handler::send_error_response(const char* error_code, const char* error_message, const char* resource_id)
+void aws_response_handler::send_error_response(const char* error_code, const char* error_message)
 {
   m_fp_chunk_encoding();
-  std::string out_error_msg = std::string(error_code) + " :" + std::string(error_message) + " :" + std::string(resource_id);
+  std::string out_error_msg = std::string(error_code) + " :" + std::string(error_message) + " :" + s->trans_id;
   error_result.resize(header_crc_size, '\0');
```
Comment (galsalomon66) [inline src/rgw/rgw_s3select.cc:221]: :+1:

File: src/rgw/rgw_s3select_private.h [hunk 4]
```diff
@@ -238,10 +236,9 @@ class RGWSelectObj_ObjStore_S3 : public RGWGetObj_ObjStore_S3
   std::function<void(void)> fp_chunked_transfer_encoding;
   int m_header_size;
 
-  const char* s3select_processTime_error = "s3select-ProcessingTime-Error";
-  const char* s3select_syntax_error = "s3select-Syntax-Error";
-  const char* s3select_resource_id = "resourcse-id";
-  const char* s3select_json_error = "json-Format-Error";
+  const char* s3select_processTime_error = "ProcessingTimeError";
```
Comment (galsalomon66) [inline src/rgw/rgw_s3select_private.h:239]: :+1:

**Comment by galsalomon66** [comment]
the shaman build had failed
https://shaman.ceph.com/builds/ceph/s3select-err/

**Comment by clwluvw** [comment]
Ah sorry, missed that in my rebase. thanks.

**Comment by clwluvw** [comment]
jenkins test make check

**Comment by galsalomon66** [comment]
https://pulpito.ceph.com/gsalomon-2024-07-17_12:35:23-rgw:verify-s3select-err-distro-default-smithi/

**Comment by clwluvw** [comment]
@galsalomon66 One of the purposes of this PR was to align the error codes with AWS style (basically CamelCase): https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html
Although there is a `ProcessingTime` error defined by AWS I think it would be great if we could keep the style unified across the platform. What do you think? I guess we can change the test. it's worth it.

**Comment by clwluvw** [comment]
@galsalomon66 Can we run the qa with https://github.com/ceph/s3-tests/pull/572?

**Comment by galsalomon66** [comment]
> @galsalomon66 One of the purposes of this PR was to align the error codes with AWS style (basically CamelCase): https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html Although there is no `ProcessingTime` error defined by AWS I think it would be great if we could keep the style unified across the platform. What do you think? I guess we can change the test. it's worth it. I'll submit a PR against S3tests

i agree with that (align with AWS-error-code system). 
but this will require a dedicated PR.


as for the change in s3test itself, it requires another change in PR(will attach that)

**Comment by galsalomon66** [comment]
_qa/suites/rgw/verify/tasks/s3tests.yaml_

```
tasks:
- s3tests:
    client.0:
      rgw_server: client.0
      force-branch: <brance-name>
      git_remote: https://github.com/<gh-user>/
```

**Comment by clwluvw** [comment]
Hmm... I guess this should be a part of teuthology to run against it... I don't see others doing it while wanted to run qa against a new branch.

**Comment by galsalomon66** [comment]
> @galsalomon66 Should be something like this: [ceph/ceph-ci@7e441b1](https://github.com/ceph/ceph-ci/commit/7e441b1fa390764a005e3f9408afb2905d7a77de)

yes
it should be on 
`qa/rgw/s3tests-branch.yaml`

**Comment by clwluvw** [comment]
I guess you need to create a branch there apply the change there and run the qa against the branch there.

**Comment by galsalomon66** [comment]
> > @galsalomon66 Should be something like this: [ceph/ceph-ci@7e441b1](https://github.com/ceph/ceph-ci/commit/7e441b1fa390764a005e3f9408afb2905d7a77de)
> 
> yes it should be on `qa/rgw/s3tests-branch.yaml`

just need to add that YAML to this PR (you already created the s3-tests branch)
upon passing teuthology, we *MUST* revert this YAML (before merging)

**Comment by clwluvw** [comment]
@galsalomon66 OK I pushed a new commit.

**Comment by galsalomon66** [comment]
https://pulpito.ceph.com/gsalomon-2024-07-18_03:25:18-rgw:verify-s3select-err-distro-default-smithi/

**Comment by galsalomon66** [comment]
https://pulpito.ceph.com/gsalomon-2024-07-18_06:33:19-rgw:verify-s3select-err-distro-default-smithi/

**Review by galsalomon66 — APPROVED** [review]

**Comment by galsalomon66** [comment]
@clwluvw 
`s3tests-branch.yaml` should revert.

https://github.com/ceph/s3-tests/pull/572 should be merged with this PR.

**Comment by clwluvw** [comment]
jenkins test ceph API tests

**Comment by clwluvw** [comment]
jenkins test api

**Comment by clwluvw** [comment]
jenkins test api

**Pull request merged**